src/state/multicall/hooks.ts

```diff
@@ -162,21 +163,31 @@ export function useSingleContractMultipleData(
 ): CallState[] {
   const fragment = useMemo(() => contract?.interface?.getFunction(methodName), [contract, methodName])
 
-  const blocksPerFetch = options?.blocksPerFetch
+  // encode callDatas
+  const callDatas = useMemo(
+    () =>
+      callInputs.every((callInput) => isValidMethodArgs(callInput)) && fragment
+        ? callInputs.map<string>((callInput) =>
+            // this type assertion is ok, because if fragment is defined, contract must be as well
+            (contract as Contract).interface.encodeFunctionData(fragment, callInput)
+          )
+        : [],
+    [callInputs, contract, fragment]
+  )
+
   const gasRequired = options?.gasRequired
+  const blocksPerFetch = options?.blocksPerFetch
 
+  // encode calls
   const calls = useMemo(
     () =>
-      contract && fragment && callInputs?.length > 0 && callInputs.every((inputs) => isValidMethodArgs(inputs))
-        ? callInputs.map<Call>((inputs) => {
-            return {
-              address: contract.address,
-              callData: contract.interface.encodeFunctionData(fragment, inputs),
-              ...(gasRequired ? { gasRequired } : {}),
-            }
-          })
-        : [],
-    [contract, fragment, callInputs, gasRequired]
+      callDatas.map<Call>((callData) => ({
+        // this type assertion is ok, because for callDatas.length to be > 0, contract must be defined
+        address: (contract as Contract).address,
+        callData,
+        ...(gasRequired ? { gasRequired } : {}),
```

Comment:
weird, why didn't we just put in `gasRequired`? it can be undefined

Reply:
yeah not sure, will change